govuk-paas/paas-rds-broker
# [#147829023] Force TLS for MySQL service instances

## What

Significant things:

- force TLS for all connections to MySQL service instances
- improve the integration tests for MySQL by testing connection and rebind permissions
- refactor the tests so that there's no duplication between Postgres and MySQL
- stop creating orphaned snapshots in the CI environment

## How to review

- It's probably best to review the commits one-by-one, reading the commit message, and viewing the diff without whitespace changes
- Confirm that the integration tests pass (status will be submitted to this PR)
- Confirm that the test coverage hasn't been reduced
- Double check my rationale about not changing URIs
- Deploy and check the acceptance test in alphagov/paas-cf#976

After merging you will need to:

- Update the submodule in alphagov/paas-rds-broker-boshrelease#40
- Delete these snapshots in the CI environment:

      aws rds describe-db-snapshots | jq '.DBSnapshots[].DBSnapshotIdentifier | select(startswith("build-test-"))'

    I haven't tested, but you can probably pipe to:

      xargs aws rds delete-db-snapshot --db-snapshot-identifier

## Who can review

Anyone
